modules/study/src/main/StudyApi.scala

```diff
@@ -268,15 +268,15 @@ final class StudyApi(
             }
           }
 
-  private def updateConceal(study: Study, chapter: Chapter, position: Position.Ref) =
+  private def updateConceal(study: Study, chapter: Chapter, position: Position.Ref): Fu[Unit] =
     chapter.conceal.so: conceal =>
-      chapter.root.lastMainlinePlyOf(position.path).some.filter(_ > conceal).so { newConceal =>
-        if newConceal >= chapter.root.lastMainlinePly then
-          for _ <- chapterRepo.removeConceal(chapter.id) yield sendTo(study.id)(_.setConceal(position, none))
-        else
-          for _ <- chapterRepo.setConceal(chapter.id, newConceal)
-          yield sendTo(study.id)(_.setConceal(position, newConceal.some))
-      }
+      if chapter.conceal.isEmpty then
```

Comment:
This will never run. `chapter.conceal.so` guarantees that `chapter.conceal` is a `Some`. Therefore `.isEmpty` will always be false.

So the following 5 LOCs can never run. Therefore the entire function never does anything.

Reply:
A naive question: why is conceal updated at all ? should not it be the decision of the study owner ?

Reply:
That code is 9 years old.

It still is obvious that conceal is intentionally and explicitely toggled off, when reaching the final position.

I can only guess that the idea is that once the entire mainline has been revealed, there is no point in hiding it again, thus it remains visible by disabling conceal.

Reply:
So you are not opposed to removing the programmatic toggle ? this solves the scenario described  by @mliebelt

Reply:
Described where?